### PR TITLE
🧹 Improved the part related to Management Group in Azure

### PR DESCRIPTION
**Pull request by HRouhani**
#### Description

This pull request introduces a Bash script designed to automate the assignment of a custom role to all subscriptions under a specified Azure Management Group. The script iterates over each subscription retrieved from the Management Group and applies the role assignment accordingly. 

#### Related issue

[[](https://github.com/mondoo-community/workspace-alnatura/issues/11)](https://github.com/mondoo-community/workspace-alnatura/issues/11)

#### Types of changes

- [ ] Functional documentation bug fix (i.e., broken link or some other busted behavior)
- [ ] New functional doc capabilities (i.e., filter search results)
- [x] New content
- [ ] Revision to existing content
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

#### Checklist

- [x] I have read the **README** document about contributing to this repo.
- [x] I have tested my changes locally and there are no issues.
- [x] All commits are signed.

### CODE REVIEW

**Comment by tas50** [comment]
Let's move tsv to the main spellcheck repo. That's going to come up soon elsewhere

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 1]
```diff
@@ -128,28 +128,31 @@ In the Azure portal, you can make sure that the created application has the requ
 
 :::
 
+
+
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:132]:
```suggestion
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 2]
```diff
@@ -132,24 +132,25 @@ In the Azure portal, you can make sure that the created application has the requ
 
 :::note
 
-The key vault permissions below are from Microsoft's "Key Vault Reader" role. The [Microsoft documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles) describes this role:
+The Key Vault permissions outlined below are based on Microsoft's "Key Vault Reader" role. The [Microsoft documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles) describes this role:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:135]:
```suggestion
The Key Vault permissions below are based on Microsoft's "Key Vault Reader" role. The [Microsoft documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles) describes this role:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 3]
```diff
@@ -132,24 +132,25 @@ In the Azure portal, you can make sure that the created application has the requ
...
 
-> _Read metadata of key vaults and its certificates, keys, and secrets. Cannot read sensitive values such as secret contents or key material. Only works for key vaults that use the 'Azure role-based access control' permission model._
+> _Read metadata of key vaults and its certificates, keys, and secrets. Cannot read sensitive values such as secret contents or key material. This role is only functional for key vaults that utilize the 'Azure role-based access control' permission model (RBAC)._
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:137]: I hate quoting their bad writing. 😆

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 4]
```diff
@@ -132,24 +132,25 @@ In the Azure portal, you can make sure that the created application has the requ
...
 
 :::
 
-Follow these steps, substituting the management group ID(s) you copied in Step A for YOUR-GROUP-ID:
+To assign this role across all subscriptions under a management group, follow these steps, replacing YOUR-GROUP-ID with your actual management group ID:
 
-1. Copy this content and paste it into a new file named `mondoo-role.json`:
+
+1. Create the Custom Role Definition
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:144]:
```suggestion
1. Create the custom role definition.
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 5]
```diff
@@ -132,24 +132,25 @@ In the Azure portal, you can make sure that the created application has the requ
...
+
+  Save the following JSON content into a file named `mondoo-role.json`:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:146]:
```suggestion
  Save this JSON content into a file named `mondoo-role.json`:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 6]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
    }
    ```
 
-2. Create a custom role:
+2. Create a Custom Role:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:185]:
```suggestion
2. Create a custom role:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 7]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
 
    ```
    az role definition create --role-definition mondoo-role.json
    ```
 
-3. Make sure you successfully created the role:
+3. Verify the custom role creation:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:191]:
```suggestion
3. Make sure you successfully created the role:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 8]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
 
    ```
    az role definition list --output json --query '[].{roleName:roleName, description:description}' --name "mondoo-role"
    ```
 
-4. Assign the created custom role to the app you registered:
+4. Check Permissions to List Subscriptions:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:197]:
```suggestion
4. Check your permissions:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 9]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
+  Confirm if your account has the required permissions to list subscriptions within the management group:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:199]:
```suggestion
  Confirm that your account has the required permissions to list subscriptions within the management group:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 10]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
+  ```
+  az account management-group subscription show-sub-under-mg --name "YOUR-GROUP-ID"
+  ```
+
+  If you encounter authorization issues, grant the necessary permissions:
+
+
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:207]:
```suggestion

```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 11]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+  ```
+  az role assignment create --assignee "YourAccountName@Domain.name" --role "Management Group Reader" --scope "/providers/Microsoft.Management/managementGroups/YOUR-GROUP-ID"
+  ```
+
+  Replace YOUR-GROUP-ID with the actual ID of your management group. Then, re-execute the command to list subscriptions and confirm access.
+
+
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:214]:
```suggestion

```
Please be sure to run the linter. :)

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 12]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+5. Assign the Custom Role Using a Script:
+
+  Use an automated script (Save the file with the name assign_role_to_subscriptions.sh) to assign this custom role to your app across all subscriptions under the management group:
+
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:219]:
```suggestion

```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 13]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
+  ```
+  az role assignment create --assignee "YourAccountName@Domain.name" --role "Management Group Reader" --scope "/providers/Microsoft.Management/managementGroups/YOUR-GROUP-ID"
+  ```
+
+  Replace YOUR-GROUP-ID with the actual ID of your management group. Then, re-execute the command to list subscriptions and confirm access.
+
+
+5. Assign the Custom Role Using a Script:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:215]:
```suggestion
5. Assign the custom role using a script:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 14]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
+  Use an automated script (Save the file with the name assign_role_to_subscriptions.sh) to assign this custom role to your app across all subscriptions under the management group:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:217]:
```suggestion
  Create a new file named `assign_role_to_subscriptions.sh` and paste this script in the file:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 15]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+      # Set the context to the current subscription
+      az account set --subscription "$SUBSCRIPTION_ID"
+      # Assign the role to the principal for the current subscription
+      az role assignment create --role "mondoo-role" --assignee $PRINCIPAL_ID --scope "/subscriptions/$SUBSCRIPTION_ID"
+  done
+   ```
+
+  Don't forget to replace YOUR-GROUP-ID with your management group ID and APP-ID with the principal ID.
+
+  Make the Script Executable:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:240]:
```suggestion
  Make the script executable:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 16]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
+   ```
+   chmod +x assign_role_to_subscriptions.sh
+   ```
+
+   and then run the script:
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:246]:
```suggestion
   Run the script:
```

File: docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx [hunk 17]
```diff
@@ -181,23 +182,77 @@ Follow these steps, substituting the management group ID(s) you copied in Step A
...
+
    ```
+   ./assign_role_to_subscriptions.sh
+   ```
+
+  These steps ensure that the custom role is systematically assigned to all subscriptions within your specified management group.
+
+
+
```
Comment (misterpantz) [inline docs/platform/infra/cloud/azure/azure-integration-scan-group.mdx:255]:
```suggestion

```

**Review by misterpantz — APPROVED** [review]

**Pull request merged**